x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -111,24 +113,86 @@ export class RuleDataClient implements IRuleDataClient {
     };
   }
 
+  createNamespacedIndexTemplate({
+    primaryNamespacedAlias,
+    secondaryNamespacedAlias,
+    namespace,
+  }: {
+    primaryNamespacedAlias: string;
+    secondaryNamespacedAlias?: string;
+    namespace?: string;
+  }): IndicesPutIndexTemplateRequest {
+    return {
+      name: primaryNamespacedAlias,
+      body: {
+        index_patterns: [`${primaryNamespacedAlias}-*`],
+        composed_of: [...this.options.componentTemplateNames],
+        template: {
+          aliases:
+            secondaryNamespacedAlias != null
+              ? {
+                  [secondaryNamespacedAlias]: {
+                    is_write_index: false,
+                  },
+                }
+              : undefined,
+          settings: {
+            'index.lifecycle': {
+              name: DEFAULT_ILM_POLICY_ID,
+              // TODO: fix the types in the ES package, they don't include rollover_alias???
+              // @ts-expect-error
+              rollover_alias: primaryNamespacedAlias,
+            },
+          },
+        },
+        _meta: {
+          namespace,
+        },
+        // By setting the priority to namespace.length, we ensure that if one namespace is a prefix of another namespace
+        // then newly created indices will use the matching template with the *longest* namespace
+        priority: namespace?.length,
+      },
+    };
+  }
+
   async createWriteTargetIfNeeded({ namespace }: { namespace?: string }) {
-    const alias = getNamespacedAlias({ alias: this.options.alias, namespace });
+    const primaryNamespacedAlias = getNamespacedAlias({ alias: this.options.alias, namespace });
 
     const clusterClient = await this.getClusterClient();
 
     const { body: aliasExists } = await clusterClient.indices.existsAlias({
-      name: alias,
+      name: primaryNamespacedAlias,
     });
 
-    const concreteIndexName = `${alias}-000001`;
+    const concreteIndexName = `${primaryNamespacedAlias}-000001`;
 
     if (!aliasExists) {
+      const secondaryNamespacedAlias =
+        this.options.secondaryAlias != null
+          ? getNamespacedAlias({ alias: this.options.secondaryAlias, namespace })
+          : undefined;
+      const template = this.createNamespacedIndexTemplate({
+        primaryNamespacedAlias,
+        secondaryNamespacedAlias,
+        namespace,
+      });
+      // TODO: need a way to update this template if/when we decide to make changes to the
+      // built in index template. Probably do it as part of updateIndexMappingsForAsset?
```

Comment:
Yeah, I think we will need it... One simple case that comes to my mind is when we add or remove a component template, we will need to update the index template as well.

Reply:
Definitely true. It may not be critical for 7.15 though since it's the first release.